frontend/app/routes/$lang+/_public+/apply+/$id+/communication-preference.tsx

```diff
@@ -79,49 +89,106 @@ export async function action({ request, params }: ActionFunctionArgs) {
+            message: t('apply:communication-preference.error-message.empty-email'),
+            path: ['email'],
+          });
+        }
         if (!emailsMatch(val.email, val.confirmEmail)) {
           ctx.addIssue({
             code: z.ZodIssueCode.custom,
-            message: 'not-the-same',
+            message: t('apply:communication-preference.error-message.not-the-same'),
+            path: ['confirmEmail'],
+          });
+        }
+        if (!z.string().email().safeParse(val.email).success) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.invalid-email'),
+            path: ['email'],
+          });
+        }
+        if (!z.string().email().safeParse(val.confirmEmail).success) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.invalid-email'),
             path: ['confirmEmail'],
           });
         }
       } else {
         if (val.emailForFuture && isEmpty(val.confirmEmailForFuture)) {
           ctx.addIssue({
             code: z.ZodIssueCode.custom,
-            message: 'empty-confirm-email',
+            message: t('apply:communication-preference.error-message.empty-confirm-email'),
             path: ['confirmEmailForFuture'],
           });
         }
         if (isEmpty(val.emailForFuture) && val.confirmEmailForFuture) {
           ctx.addIssue({
             code: z.ZodIssueCode.custom,
-            message: 'empty-email',
+            message: t('apply:communication-preference.error-message.empty-email'),
             path: ['emailForFuture'],
           });
         }
         if (!emailsMatch(val.emailForFuture, val.confirmEmailForFuture)) {
           ctx.addIssue({
             code: z.ZodIssueCode.custom,
-            message: 'not-the-same',
+            message: t('apply:communication-preference.error-message.not-the-same'),
             path: ['confirmEmailForFuture'],
           });
         }
+        if (!isEmpty(val.emailForFuture) || !isEmpty(val.confirmEmailForFuture)) {
+          if (!z.string().email().safeParse(val.emailForFuture).success) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: t('apply:communication-preference.error-message.invalid-email'),
+              path: ['emailForFuture'],
+            });
+          }
+          if (!z.string().email().safeParse(val.confirmEmailForFuture).success) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: t('apply:communication-preference.error-message.invalid-email'),
+              path: ['confirmEmailForFuture'],
+            });
+          }
+        }
       }
     });
 
-  const formData = Object.fromEntries(await request.formData());
-  const parsedDataResult = formSchema.safeParse(formData);
+  const formData = await request.formData();
+  const data = {
```

Comment:
Handle `zod.optional()` fiels like this.

```suggestion
  email: formData.get('email') ? String(formData.get('email')) : undefined,
```

Reply:
change made